src/transformers/modeling_flax_utils.py

```diff
@@ -657,6 +657,20 @@ def from_pretrained(
                 f"You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference."
             )
 
+        # dictionary of key: bools that establish whether each parameter is in jnp.float32
+        param_dtypes = jax.tree_map(lambda x: x.dtype != jnp.float32, state)
+        # extract keys of parameters not in jnp.float32
+        downcast_params = [k for k in param_dtypes if param_dtypes[k]]
```

Comment:
(nit) maybe name it as `half_precision_params`

Reply:
Have modified the code to generate two different lists: one for fp16 params and another for bf16 params. The warning message then specifies the erroneous dtype of the model weights loaded. I think this is more informative than simply saying the weights are in a dtype different to fp32.